certbot/cli.py

```diff
@@ -709,6 +709,11 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):  # pylint: dis
         help="With the register verb, indicates that details associated "
              "with an existing registration, such as the e-mail address, "
              "should be updated, rather than registering a new account.")
+    helpful.add(
+        "register", "--deactivate", action="store_true",
+        help="Irrevocably deactivate your account. Certificates associated "
+             "with your account can still be revoked, but they can not be "
+             "renewed.")
```

Comment:
Authorization reuse is the biggest reason.

After successfully performing a DV challenge, the ACME server has a valid authorization object saying that it may issue a certificate to the owner of the account whose request created the authorization. This authorization does not expire immediately, even after the certificate issued. This means that the owner of an account key can issue certificates for that domain without performing DV challenges until the authorization expires. Because of this, if someone steals your account key, they can issue a certificate for your domain.

Let's Encrypt does this proactively, reusing authorizations even if you request a new one ([source](https://community.letsencrypt.org/t/upcoming-change-valid-authz-reuse/16982)). This has lead to some [interesting conversations](https://news.ycombinator.com/item?id=12787735), but behavior like this is common practice for CAs and is completely valid in CAB forum requirements.